e2e/cypress/integration/channel_sidebar/channel_sidebar_spec.js

```diff
@@ -0,0 +1,130 @@
+
+        // * Verify that both Off Topic and Town Square are read
+        cy.get('.SidebarChannel:not(.unread):contains(Off-Topic)').should('be.visible');
+        cy.get('.SidebarChannel:not(.unread):contains(Town Square)').should('be.visible');
+
+        // # Have another user post in the Off Topic channel
+        cy.apiGetChannelByName(teamName, 'off-topic').then((response) => {
+            expect(response.status).to.equal(200);
+
+            const channel = response.body;
+            cy.postMessageAs({sender: sysadmin, message: 'Test', channelId: channel.id});
+        });
+
+        // * Verify that Off Topic is unread and Town Square is read
+        cy.get('.SidebarChannel.unread:contains(Off-Topic)').should('be.visible');
+        cy.get('.SidebarChannel:not(.unread):contains(Town Square)').should('be.visible');
+    });
+
+    it('should remove channel from sidebar after leaving it', () => {
+        // # Start with a new team
+        const teamName = `team-${getRandomInt(999999)}`;
+        cy.createNewTeam(teamName, teamName);
+
+        // * Verify that we've switched to the new team
+        cy.get('#headerTeamName').should('contain', teamName);
+
+        // # Switch to Off Topic
+        cy.visit(`/${teamName}/channels/off-topic`);
+
+        // # Wait for the channel to change
+        cy.get('#channelHeaderTitle').should('contain', 'Off-Topic');
+
+        // # Click on the channel menu and select Leave Channel
+        cy.get('#channelHeaderTitle').click();
+        cy.get('#channelLeaveChannel').click();
+
+        // * Verify that we've switched to Town Square
+        cy.url().should('include', `/${teamName}/channels/town-square`);
+        cy.get('#channelHeaderTitle').should('contain', 'Town Square');
+
+        // * Verify that Off Topic has disappeared from the sidebar
+        cy.get('.SidebarChannel:contains(Off-Topic)').should('not.exist');
+    });
+
+    it('MM-23239 should remove channel from sidebar after deleting it', () => {
+        // # Start with a new team
+        const teamName = `team-${getRandomInt(999999)}`;
+        cy.createNewTeam(teamName, teamName);
+
+        // * Verify that we've switched to the new team
+        cy.get('#headerTeamName').should('contain', teamName);
+
+        // # Switch to Off Topic
+        cy.visit(`/${teamName}/channels/off-topic`);
+
+        // # Wait for the channel to change
+        cy.get('#channelHeaderTitle').should('contain', 'Off-Topic');
+
+        // # Click on the channel menu and select Leave Channel
+        cy.get('#channelHeaderTitle').click();
+        cy.get('#channelArchiveChannel').click();
+
+        // * Verify that we've switched to Town Square
+        cy.url().should('include', `/${teamName}/channels/town-square`);
+        cy.get('#channelHeaderTitle').should('contain', 'Town Square');
+
+        // * Verify that Off Topic has disappeared from the sidebar
+        cy.get('.SidebarChannel:contains(Off-Topic)').should('not.exist');
```

Comment:
This failed since off-topic is still in the sidebar after archived.
![Screen Shot 2020-03-18 at 4 49 07 AM](https://user-images.githubusercontent.com/5334504/76900441-4fd5f980-68d4-11ea-8e6e-7f583b4f1d18.png)

Reply:
That's probably failing because you have the View Archived Channels setting still enabled. I'll add that to the `testWithConfig`